contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/BasicFetchStrategy.java

```diff
@@ -64,21 +63,17 @@ public BasicFetchStrategy(HdfsFetcher fetcher, FileSystem fs, HdfsCopyStats stat
      * 'checkSumType' computed and returned. In case an error occurs during such
      * a copy, we do a retry for a maximum of NUM_RETRIES
      *
-     * @param fs
-     *            Filesystem used to copy the file
      * @param source
      *            Source path of the file to copy
      * @param dest
      *            Destination path of the file on the local machine
-     * @param stats
-     *            Stats for measuring the transfer progress
      * @param checkSumType
      *            Type of the Checksum to be computed for this file
      * @return A Checksum (generator) of type checkSumType which contains the
      *         computed checksum of the copied file
      * @throws IOException
      */
-    private CheckSum copyFileWithCheckSum(HdfsFile source, File dest, CheckSumType checkSumType) throws IOException {
+    private CheckSum copyFileWithCheckSum(HdfsFile source, File dest, CheckSumType checkSumType) throws Throwable {
```

Comment:
Again, I agree with you that dealing with ```Throwable```s is generally a bad idea, but this is how the original code was, before your recent refactoring:

https://github.com/voldemort/voldemort/commit/168cb69f3732ac7164f2e66ae8f1ac7239e62323#diff-bed2a573ebc48c26a9eacc4cccd7ef90R180

So my change is actually bringing back the ```Throwable``` that was already there, rather than introducing it in the first place.

I will need to search our production logs to investigate what is it that we actually catch in this block. If some of the HDFS APIs we use actually throw ```Throwable```s and we rely on that for retrying, then we may, unfortunately, need to keep that in there.

-F